charms/kfp-profile-controller/src/charm.py

```diff
@@ -0,0 +1,221 @@
+                            },
+                        ],
+                        "envConfig": deployment_env,
+                        "volumeConfig": [
+                            {
+                                "name": "kubeflow-pipelines-profile-controller-code",
+                                "mountPath": "/hooks",
+                                "configMap": {
+                                    "name": "kubeflow-pipelines-profile-controller-code",
+                                    "files": [
+                                        {
+                                            "key": "sync.py",
+                                            "path": "sync.py",
+                                        },
+                                    ],
+                                },
+                            },
+                        ],
+                    },
+                ],
+            },
+            k8s_resources={
+                "kubernetesResources": {
+                    "customResources": {
+                        # Define the metacontroller that manages the deployments for each user
+                        "compositecontrollers.metacontroller.k8s.io": [
+                            {
+                                "apiVersion": "metacontroller.k8s.io/v1alpha1",
+                                "kind": "CompositeController",
+                                "metadata": {
+                                    "name": "kubeflow-pipelines-profile-controller",
+                                },
+                                "spec": {
+                                    "childResources": [
+                                        {
+                                            "apiVersion": "v1",
+                                            "resource": "secrets",
+                                            "updateStrategy": {"method": "OnDelete"},
+                                        },
+                                        {
+                                            "apiVersion": "v1",
+                                            "resource": "configmaps",
+                                            "updateStrategy": {"method": "OnDelete"},
+                                        },
+                                        {
+                                            "apiVersion": "apps/v1",
+                                            "resource": "deployments",
+                                            "updateStrategy": {"method": "InPlace"},
+                                        },
+                                        {
+                                            "apiVersion": "v1",
+                                            "resource": "services",
+                                            "updateStrategy": {"method": "InPlace"},
+                                        },
+                                        # TODO: This only works if istio is available.  Disabled
```

Comment:
If this is important for the service too work we should probably just do a hard check for istio through a relation.